winsup/cygwin/msys2_path_conv.cc

```diff
@@ -358,6 +358,40 @@ path_type find_path_start_and_type(const char** src, int recurse, const char* en
         return NONE;
     }
 
+    /*
+     * Discern between Git's `<rev>:<path>`, SCP's `<host>:<path>` pattern
+     * (which is not a path list but may naïvely look like one) on the one
+     * hand, and path lists starting with `/<path>`, `./<path>` or `../<path>`
+     * on the other hand.
+     */
+    bool potential_path_list = *it == '/' ||
+	    (*it == '.' &&
+		(it[1] == ':' || it[1] == '/' ||
+		 (it[1] == '.' &&
+			(it[2] == ':' || it[2] == '/'))));
+
```

Comment:
```suggestion
    /*
     * Discern between Git's `<rev>:<path>`, SCP's `<host>:<path>` pattern
     * (which is not a path list but may naively look like one) on the one
     * hand, and path lists starting with `/<path>`, `./<path>` or `../<path>`
     * on the other hand.
     */
    bool potential_path_list = *it == '/' ||
        (*it == '.' &&
         (it + 1 == end || it[1] == '\0' || it[1] == ':' || it[1] == '/' ||
          (it[1] == '.' &&
           (it + 2 == end || it[2] == '\0' || it[2] == ':' || it[2] == '/'))));

```
* Watch for end-of-string / end of input.
* This file mostly doesn't use tabs.
* Let's not introduce a character `ï` that isn't encoded the same in iso-latin1 and utf-8 (there are no others in this file yet).

Reply:
@reneparis, what do you think?

Reply:
@ReinhardBiegelIntech, what do you think?

(This replaces a deleted comment which tagged @ reneparis. Apologies, René.)

Reply:
I checked your code and agree on your suggestions - so let's wait for reinhard (currently on vacation) to implement it as soon as he's back.

Reply:
Thanks for your feedback, @bustercopley. Suggestions look good to me, so I updated the PR accordingly. Also committed to [test suite](https://github.com/msys2/msys2-tests/pull/67).

As the changes affect code originally written bei @dscho, I'd also like to get his feedback here.